indexing-service/src/main/java/org/apache/druid/indexing/common/task/batch/parallel/PartialHashSegmentGenerateTask.java

```diff
@@ -158,17 +164,24 @@ private HashPartitionStat createPartitionStat(TaskToolbox toolbox, DataSegment s
     );
   }
 
-  private Map<Interval, Pair<ShardSpecFactory, Integer>> createShardSpecs()
+  /**
+   * Creates shard specs based on the given configurations. The return value is a map between intervals created
+   * based on the segment granularity and the shard specs to be created.
+   * Note that the shard specs to be created is a pair of {@link PartialShardSpec} and number of segments per interval
+   * and filled only when {@link #isGuaranteedRollup} = true. Otherwise, the return value contains only the set of
+   * intervals generated based on the segment granularity.
+   */
+  public static HashPartitionAnalysis createHashPartitionAnalysisFromPartitionsSpec(
```

Comment:
Is it worth adding a unit test for this method now?

Reply:
Added.